synapse/handlers/message.py

```diff
@@ -1930,6 +1934,27 @@ async def _notify() -> None:
 
         return persisted_events[-1]
 
+    async def is_admin_redaction(
+        self, event_type: str, sender: str, redacts: Optional[str]
+    ) -> bool:
+        is_admin_redaction = False
+        if event_type == EventTypes.Redaction:
+            assert redacts is not None
+
+            original_event = await self.store.get_event(
+                redacts,
+                redact_behaviour=EventRedactBehaviour.as_is,
+                get_prev_content=False,
+                allow_rejected=False,
+                allow_none=True,
+            )
+
+            is_admin_redaction = bool(
+                original_event and sender != original_event.sender
+            )
+
+        return is_admin_redaction
```

Comment:
This should be the same and is easier to read IMO.

```suggestion
    async def is_admin_redaction(
        self, event_type: str, sender: str, redacts: Optional[str]
    ) -> bool:
        if event_type != EventTypes.Redaction:
            return False

        assert redacts is not None

        original_event = await self.store.get_event(
            redacts,
            redact_behaviour=EventRedactBehaviour.as_is,
            get_prev_content=False,
            allow_rejected=False,
            allow_none=True,
        )

        return bool(
            original_event and sender != original_event.sender
        )
```

Regardless of which you like, can we add a docstring? 😄